src/Uno.UWP/ApplicationModel/Contacts/ContactReader.Android.cs

```diff
@@ -0,0 +1,436 @@
+				entry.HonorificNamePrefix = subCursor.GetString(3) ?? "";
+				entry.HonorificNameSuffix = subCursor.GetString(5) ?? "";
+				entry.DisplayNameOverride = subCursor.GetString(0) ?? "";
+
+				entry.YomiGivenName = subCursor.GetString(6) ?? "";
+				entry.YomiFamilyName = subCursor.GetString(7) ?? "";
+
+				if (!searchFound && _queryOptions.TextSearch.Fields.HasFlag(ContactQuerySearchFields.Name))
+				{
+					if (entry.MiddleName.Contains(_queryOptions.TextSearch.Text, StringComparison.OrdinalIgnoreCase) ||
+							entry.LastName.Contains(_queryOptions.TextSearch.Text, StringComparison.OrdinalIgnoreCase) ||
+							entry.FirstName.Contains(_queryOptions.TextSearch.Text, StringComparison.OrdinalIgnoreCase) ||
+							entry.HonorificNamePrefix.Contains(_queryOptions.TextSearch.Text, StringComparison.OrdinalIgnoreCase) ||
+							entry.HonorificNameSuffix.Contains(_queryOptions.TextSearch.Text, StringComparison.OrdinalIgnoreCase) ||
+							entry.DisplayName.Contains(_queryOptions.TextSearch.Text, StringComparison.OrdinalIgnoreCase) ||
+							entry.YomiGivenName.Contains(_queryOptions.TextSearch.Text, StringComparison.OrdinalIgnoreCase) ||
+							entry.YomiFamilyName.Contains(_queryOptions.TextSearch.Text, StringComparison.OrdinalIgnoreCase))
+
+					{
+						searchFound = true;
+					}
+				}
+
+
+			}
+			subCursor.Close();
+			return searchFound;
+		}
+
+		private bool ReadContactAddresses(Contact entry, int contactId, Android.Content.ContentResolver contentResolver)
+		{
+			var contentUri = Android.Provider.ContactsContract.Data.ContentUri;
+			if (contentUri is null)
+			{
+				throw new InvalidOperationException("Windows.ApplicationModel.Contacts.ContactReader.ReadBatchInternal, ContentUri3 is null (impossible)");
+			}
+
+			var subCursor = _contentResolver?.Query(
+							contentUri,
+							new string[] { "data2", "data4", "data7", "data8", "data9", "data10" }, //null,// all columns
+																									// ContactsContract.Data.RAW_CONTACT_ID + " = ? AND " + ContactsContract.Data.MIMETYPE + " = ?",
+							"raw_contact_id = ? AND mimetype = ?",
+							new string[] { contactId.ToString(), Android.Provider.ContactsContract.CommonDataKinds.StructuredPostal.ContentItemType },
+							null);   // default order
+			if (subCursor is null)
+			{
+				throw new InvalidOperationException("Windows.ApplicationModel.Contacts.ContactReader.ReadBatchInternal, subCursor3 is null (impossible)");
+			}
+
+			bool searchFound = false;
+
+			for (int itemGuard = 10; itemGuard > 0 && subCursor.MoveToNext(); itemGuard--)
+			{
+				var itemEntry = new ContactAddress();
+				itemEntry.StreetAddress = subCursor.GetString(1) ?? "";    // we defined columns while opening cursor, so we know what data is in which columns
+				itemEntry.Region = subCursor.GetString(3) ?? "";
+				itemEntry.PostalCode = subCursor.GetString(4) ?? "";
+				itemEntry.Locality = subCursor.GetString(2) ?? "";
+				//itemEntry.Description = subCursor.GetString(columnD4);

```

Comment:
commented? still needed for the Description?

Reply:
Added comment:
// *TODO* we don't set this field value, but field is not marked as NotImplemented, so (in theory) we should not ignore this field